modules/video/src/bgfg_gaussmix2.cpp

```diff
@@ -693,11 +693,16 @@ class MOG2Invoker : public ParallelLoopBody
                 //go through all modes
                 //////
 
-                //renormalize weights
-                totalWeight = 1.f/totalWeight;
+                // Renormalize weights. In the special case that the pixel does
+                // not agree with any modes, set weights to zero (a new mode will be added below).
+                float invWeight = 0.f;
+                if (fabs(totalWeight) > DBL_EPSILON) {
```

Comment:
could you use std::abs() here and comparison with FLT_EPSILON or such? We don't want to use double precision unless it's not absolutely necessary.